### PR TITLE
[DataPipe] Disable Typing for DataPipe before branch cut

### PR DESCRIPTION
**Pull request by ejguan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#72123 [DataPipe] Disable Typing for DataPipe before branch cut**

There is a bug to fix the typing system in DataPipe, which would take more than 1 week to fix. I will follow up on it later this month. As branch cut is today, add this PR to disable typing to make sure release works.

Besides, another fix for typing is needed for the sake of Python 3.10.

Differential Revision: [D33920610](https://our.internmc.facebook.com/intern/diff/D33920610)

### CODE REVIEW

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33920610).

File: test/test_datapipe.py [hunk 1]
```diff
@@ -1543,6 +1544,18 @@ class InvalidData(Generic[T_co], NamedTuple):
...
+    def test_isinstance(self):
+        class A(IterDataPipe):
+            pass
+
+        class B(IterDataPipe):
+            pass
+
+        a = A()
+        self.assertTrue(isinstance(a, A))
+        self.assertFalse(isinstance(a, B))
```
Comment (ejguan) [inline test/test_datapipe.py:1556]: The bug is related `MRO` of the metaclass. This test won't pass with typing.

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33920610).

**Review by VitalyFedyunin — APPROVED** [review]

**Review by NivekT — APPROVED** [review]

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33920610).

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33920610).

**Pull request closed without merging**